src/Bulk/Action/UpdateDocument.php

```diff
@@ -51,6 +51,7 @@ public function setScript(AbstractScript $script): AbstractDocument
 
             if (!empty($upsert)) {
                 $source['upsert'] = $upsert;
+                $source['scripted_upsert'] = true;
```

Comment:
To keep the option option to also set it to false I wonder if we should pass a second param to the setScript method.

Reply:
Problem is that the `setScript` is called in the `AbstractDocument`. Don't think it makes much sense to mix the document and script here (in this case, adding one specific parameter of script).

Maybe adding it in the `AbstractScript` would be better? 
So it becomes somehow similar to the implementation of `doc_as_upsert`.

Reply:
If we add it to AbstractScript, would it be also useful to the other classes or it doesn't make sense there?

Didn't fully get your comment related to `doc_as_upsert`. Can you share more?

Reply:
Well, the AbstractScript is used in other places, not only for bulk operations. I didn't reviewed all the cases, but I'm pretty confident that the `scripted_upsert` won't be expected for all of them. Having said that, we'd be using that parameter only within the bulk context.

What I meant with the `doc_as_upsert` is regarding the implementation above (lines 18-27 of the same file), where we check for the value of `$document->getDocAsUpsert()` and only then add it to the request.

Reply:
@dsgrillo Sorry that this got stalled on my end. As you describe above, adding it to AbstractScript might not be ideal as it is not expected for all of them. So we are left with how to add it here. 

What you propose with `$document->getDocAsUpsert()` sounds like a good path forward. If I understand this correct it means it would get the "value" from the doc itself and we would not need an additional param? I guess this requires that the document is set first?